programs/smart-wallet/src/instructions/buffer_execute_ix.rs

```diff
@@ -0,0 +1,59 @@
+//! Executes an instruction off of the [InstructionBuffer].
+
+use crate::*;
+
+use anchor_lang::solana_program::program::invoke_signed;
+
+#[derive(Accounts)]
+pub struct ExecuteBufferIX<'info> {
+    #[account(mut)]
+    /// The [InstructionBuffer].
+    pub buffer: Account<'info, InstructionBuffer>,
+    /// The buffer's [SmartWallet]
+    pub smart_wallet: Account<'info, SmartWallet>,
+    /// The [InstructionBuffer::executor].
+    pub executor: Signer<'info>,
+}
+
+pub fn handler<'info>(
+    ctx: Context<'_, '_, '_, 'info, ExecuteBufferIX<'info>>,
+    bundle_index: u8,
+) -> Result<()> {
+    let smart_wallet = &ctx.accounts.smart_wallet;
+    let wallet_seeds: &[&[&[u8]]] = &[&[
+        b"GokiSmartWallet" as &[u8],
+        &smart_wallet.base.to_bytes(),
+        &[smart_wallet.bump],
+    ]];
+
+    let buffer = &mut ctx.accounts.buffer;
+    let mut bundle = unwrap_opt!(buffer.get_bundle(bundle_index), BufferBundleNotFound);
+    invariant!(!bundle.is_executed(), BufferBundleExecuted);
+
+    let ix = &bundle.instructions[usize::from(bundle.exec_count)];
```

Comment:
Let's have bundles be executed atomically as one transaction. No exec_count

Reply:
We should keep the instruction small.